### PR TITLE
Clean up docstring and initialization of Autoencoder_torch

### PR DESCRIPTION
**Pull request by Lucew**
This pull request is a partial update to #375

As a first step, I cleaned the Autoencoder_Torch Docstring and Initialization in order to make clear and mention, that the learning rate is a valid parameter for the model. 
In the process, I also cleaned the code in the initialization and changed some wrong links (for tensorflow) to the corresponding Pytorch ones.

I did not alter any functionality, as learning rate was already supported (but very hidden).

With PyTorch it is difficult to pass an optimizer (with several changed parameters) as an instance since you need to tie it immediately to the model parameters.
Unfortunately, the model can only be initialized once the amount of features is known, which is only happening once the package user calls the fit function. This is also where the Optimizer gets initialized.

<Details>

https://github.com/yzhao062/pyod/blob/master/pyod/models/auto_encoder_torch.py#L259-L299

</Details>

I ran the tests corresponding to the Autoencoder_PyTorch locally and all tests passed on my own fork.

Expect more PRs once I have time, but for now at least the PyTorch version of Autoencoder has an improved docstring mentioning learning rate.

### All Submissions Basics:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Have you checked all [Issues](../../issues) to tie the PR to a specific one?

### All Submissions Cores:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable? **not applicable**
* [X] Have you successfully ran tests with your changes locally?
* [X] Does your submission pass tests, including CircleCI, Travis CI, and AppVeyor?
* [X] Does your submission have appropriate code coverage? The cutoff threshold is 95% by Coversall.

### CODE REVIEW

**Comment by yzhao062** [comment]
tests are enabled to run. If nothing breaks I will merge it right away. Thanks for improving the doc

**Comment by Lucew** [comment]
Thanks for your quick feedback!
